docs/basic-docs/atomicdex-api-20-dev/trezor_integration.md

```diff
@@ -0,0 +1,1924 @@
+
+<collapse-text hidden title="Response">
+
+```json
+{
+    "mmrpc": "2.0",
+    "result": {
+        "status": "Error",
+        "details": {
+            "error": "Found unexpected device. Please re-initialize Hardware wallet",
+            "error_path": "lib.common_impl.coin_balance.utxo_common.hd_pubkey.hw_ctx",
+            "error_trace": "lib:93] common_impl:46] coin_balance:304] utxo_common:163] hd_pubkey:176] hw_ctx:149]",
+            "error_type": "HwError",
+            "error_data": "FoundUnexpectedDevice"
+        }
+    },
+    "id": null
+}
+```
+
+</collapse-text>
+
+</div>
+
+
+# task\_enable\_utxo\_user_action
+
+If the `task::enable_utxo::status` returns `UserActionRequired`, we need to use the `task::enable_utxo::user_action` method to enter our PIN
+
+
+#### Arguments
+
+| Parameter                    | Type            | Description                                                                 |
+| ---------------------------- | --------------- | --------------------------------------------------------------------------- |
+| task_id                      | integer         | The identifying number returned when initiating the initialisation process. |
+| user_action                  | object          | Object containing the params below                                          |
+| user_action.action_type      | string          | Will be `TrezorPin` for this method                                         |
+| user_action.pin              | string (number) | When the Trezor device is displaying a grid of numbers for PIN entry, this param will contain your Trezor pin, as mapped through your keyboard numpad. See the image below for more information.  |
+
+
+<div style="margin: 2rem; text-align: center; width: 80%">
+    <img src="/api_images/trezor_pin.png" />
+</div>
+
+
+#### Response
+
+| Parameter             | Type         | Description                 |
+| --------------------- | -------------| --------------------------- |
+| result                | string       | The outcome of the request. |
+
+
+#### :pushpin: Examples
+
+#### Command
+
+```bash
+curl --url "http://127.0.0.1:7783" --data "{
+    \"userpass\": \"YOUR_PASS\",
+    \"mmrpc\": \"2.0\",
+    \"method\": \"task::enable_utxo::user_action\",
+    \"params\": {
+        \"task_id\": 0,
+        \"user_action\": {
+            \"action_type\": \"TrezorPin\",
+            \"pin\": \"862743\"
+        }
+    }
+}"
+```
+
+<div style="margin-top: 0.5rem;">
+
+<collapse-text hidden title="Response">
+
+#### Response (success)
+
+```json
+{
+    "mmrpc":"2.0",
+    "result":"success",
+    "id":null
+}
+```
+
+</collapse-text>
+
+</div>
+
+
+# task\_enable\_qtum\_init
+
+UTXO coins are activated using this method. For UTXO coins, refer to [task::enable_utxo::init]("#task_enable_utxo_init")
+
+#### Arguments
+
+| Parameter                     | Type    | Description                                                                     |
```

Comment:
I think in general where request payloads are the same (or 80% same with the remaining optional) it is good to have a single method that can be used for both. It reduces confusion from user and overhead to create maintain docs. 

IMO the ideal solution would be a single "enable_coin" method which can be used on anything, with API + coins file able to determine which underlying method to use. Another enhancement would be allowing user to define [coins_config.json](https://github.com/KomodoPlatform/coins/blob/master/utils/coins_config.json) and derive the rpc_nodes / electrums etc from there - though I understand this would be a fairly big job and maybe better to plan towards for v3 of the API

Reply:
I can't answer this suggestion correctly as I didn't work on `enable V2` feature, so I'll just reference to this comment https://github.com/KomodoPlatform/atomicDEX-API/issues/1006#issuecomment-960594323 and to your comment https://github.com/KomodoPlatform/atomicDEX-API/issues/1006#issuecomment-960663488 🙂